modules/basegl/processors/imageprocessing/imagescaling.cpp

```diff
@@ -101,37 +100,48 @@ void ImageScaling::process() {
 
 void ImageScaling::propagateEvent(Event* event, Outport* source) {
     if (event->hasVisitedProcessor(this)) return;
-    event->markAsVisited(this);
 
-    invokeEvent(event);
-    if (event->hasBeenUsed()) return;
+    if ((event->hash() == ResizeEvent::chash())) {
+        // cache size of the resize event
+        lastValidOutputSize_ = event->getAs<ResizeEvent>()->size();
 
-    if (event->hash() == ResizeEvent::chash()) {
-        if (inport_.isConnected()) {
-            ResizeEvent e(calcInputImageSize());
-            inport_.propagateEvent(&e);
-        }
-    } else {
-        bool used = event->hasBeenUsed();
-        if (event->shouldPropagateTo(&inport_, this, source)) {
-            inport_.propagateEvent(event);
-            used |= event->hasBeenUsed();
+        if (enabled_) {
```

Comment:
Move to outer it

Reply:
but the size of the resize event must be cached independent of whether the processor is enabled or not.